spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -0,0 +1,170 @@
+ * If no applicable value is found in the session configuration, this classes uses the table metadata.
+ * <p>
+ * Note this class is NOT meant to be serialized and sent to executors.
+ */
+public class SparkWriteConf {
+
+  private final Table table;
+  private final RuntimeConfig sessionConf;
+  private final Map<String, String> writeOptions;
+
+  public SparkWriteConf(SparkSession spark, Table table, Map<String, String> writeOptions) {
+    this.table = table;
+    this.sessionConf = spark.conf();
+    this.writeOptions = writeOptions;
+  }
+
+  public boolean checkNullability() {
+    String writeOptionValue = writeOptions.get(SparkWriteOptions.CHECK_NULLABILITY);
+    if (writeOptionValue != null) {
+      return Boolean.parseBoolean(writeOptionValue);
+    }
+
+    String sessionConfValue = sessionConf.get(SparkSQLConfigs.CHECK_NULLABILITY, null);
+    if (sessionConfValue != null) {
+      return Boolean.parseBoolean(sessionConfValue);
+    }
+
+    return true;
+  }
+
+  public boolean checkOrdering() {
+    String writeOptionValue = writeOptions.get(SparkWriteOptions.CHECK_ORDERING);
+    if (writeOptionValue != null) {
+      return Boolean.parseBoolean(writeOptionValue);
+    }
+
+    String sessionConfValue = sessionConf.get(SparkSQLConfigs.CHECK_ORDERING, null);
+    if (sessionConfValue != null) {
+      return Boolean.parseBoolean(sessionConfValue);
+    }
+
+    return true;
+  }
+
+  /**
+   * Enables writing a timestamp with time zone as a timestamp without time zone.
+   * <p>
+   * Generally, this is not safe as a timestamp without time zone is supposed to represent the wall-clock time,
+   * i.e. no matter the reader/writer timezone 3PM should always be read as 3PM,
+   * but a timestamp with time zone represents instant semantics, i.e. the timestamp
+   * is adjusted so that the corresponding time in the reader timezone is displayed.
+   * <p>
+   * When set to false (default), an exception must be thrown if the table contains a timestamp without time zone.
+   *
+   * @return boolean indicating if writing timestamps without timezone is allowed
+   */
+  public boolean handleTimestampWithoutZone() {
+    String writeOptionValue = writeOptions.get(SparkWriteOptions.HANDLE_TIMESTAMP_WITHOUT_TIMEZONE);
+    if (writeOptionValue != null) {
+      return Boolean.parseBoolean(writeOptionValue);
+    }
+
+    String sessionConfValue = sessionConf.get(SparkSQLConfigs.HANDLE_TIMESTAMP_WITHOUT_TIMEZONE, null);
+    if (sessionConfValue != null) {
+      return Boolean.parseBoolean(sessionConfValue);
+    }
+
+    return false;
+  }
+
+  public String overwriteMode() {
+    String overwriteMode = writeOptions.get("overwrite-mode");
+    return overwriteMode != null ? overwriteMode.toLowerCase(Locale.ROOT) : null;
+  }
+
+  public String wapId() {
+    return sessionConf.get("spark.wap.id", null);
+  }
+
+  public FileFormat dataFileFormat() {
+    String writeOptionValue = writeOptions.get(SparkWriteOptions.WRITE_FORMAT);
```

Comment:
this is more of the code I think we can simplify by making a common

getConfiguredString(String name, String default)

Reply:
But that will require us to read the default before reading the options value. 

I am thinking maybe `Optional<String> getOption(String optionName)`, and then we can chain it with table property default, just a thought.